### PR TITLE
fix: descriptions not re-rendered when slots is updated (#2275)

### PR DESCRIPTION
**Pull request by msidolphin**
Please make sure these boxes are checked before submitting your PR, thank you!
bugfix #2275 
* [x] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/descriptions/src/index.vue [hunk 1]
```diff
@@ -96,7 +73,7 @@ export default defineComponent({
       return node
     }
 
-    const rows = computed(() => {
+    const getRows = () => {
```
Comment (HerringtonDarkholme) [inline packages/descriptions/src/index.vue:76]: We can have a easier fix: just expose `getRows` and invoke it in template

File: packages/descriptions/src/index.vue [hunk 2]
```diff
@@ -1,28 +1,5 @@
...
-      </div>
-      <div class="el-descriptions__extra">
-        <slot name="extra">{{ extra }}</slot>
-      </div>
-    </div>
-
-    <div class="el-descriptions__body">
-      <table :class="[{'is-bordered': border}, descriptionsSize ? `el-descriptions--${descriptionsSize}` : '']">
-        <tbody>
-          <template v-for="(row, index) in rows" :key="index">
```
Comment (HerringtonDarkholme) [inline packages/descriptions/src/index.vue:15]: Just invoke `getRows` here

**Review by HerringtonDarkholme — CHANGES_REQUESTED** [review]
A proper test case is also needed. Could you add one?

File: packages/descriptions/src/index.vue (hunk 1 above)
Comment (msidolphin) [reply packages/descriptions/src/index.vue:76]: You are right, the reason I do this is because I think that using render function is more flexible

**Comment by msidolphin** [comment]
@HerringtonDarkholme I will add proper test case tomorrow

File: packages/descriptions/src/index.vue (hunk 1 above)
Comment (HerringtonDarkholme) [reply packages/descriptions/src/index.vue:76]: Sadly render function is much harder for even component library authors. Avoid unless necessary.

Comment (msidolphin) [reply packages/descriptions/src/index.vue:76]: Okay, I see. I'll fix it

**Review by HerringtonDarkholme — APPROVED** [review]

**Pull request merged**